agent/agent-gc-monitor/gc-monitor-core/src/main/java/com/microsoft/gcmonitor/collectors/JMXGarbageCollector.java

```diff
@@ -0,0 +1,100 @@
+package com.microsoft.gcmonitor.collectors;
+
+import com.microsoft.gcmonitor.GCEventConsumer;
+import com.microsoft.gcmonitor.MemoryManagement;
+import com.microsoft.gcmonitor.UnableToMonitorMemoryException;
+import com.microsoft.gcmonitor.garbagecollectors.GarbageCollector;
+import com.microsoft.gcmonitor.garbagecollectors.GarbageCollectorStats;
+import com.microsoft.gcmonitor.garbagecollectors.GarbageCollectors;
+import com.microsoft.gcmonitor.memorypools.MemoryPool;
+
+import javax.management.JMX;
+import javax.management.MBeanServerConnection;
+import javax.management.Notification;
+import javax.management.ObjectName;
+import javax.management.openmbean.CompositeData;
+import java.lang.management.GarbageCollectorMXBean;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Implementation of a GarbageCollectorStats that provides access to the stats via a MxBean
+ */
+public class JMXGarbageCollector implements GarbageCollectorStats {
```

Comment:
to not get this confused with `GarbageCollector`
```suggestion
public class JmxGarbageCollectorStats implements GarbageCollectorStats {
```